loosens the solidus core version to < 2.0

sherryson opened the pull request:
this is required for apps using this gem to upgrade to higher versions of Solidus (e.g. 1.4.2)
also removes Solidus from the Gemfile - seems odd to be based off the master branch and was preventing me from being able to bundle (lots of conflicts)

sherryson committed:
loosens the solidus core version to < 2.0

this is required for apps using this gem to upgrade to higher versions of Solidus (e.g. 1.4.2)
also removes Solidus from the Gemfile - seems odd to be based off the master branch and was preventing me from being able to bundle (lots of conflicts)

sherryson commented:
@braidn not sure if you might have any insights here, but i'm really struggling to run the specs. specifically can't run `bundle exec rake test_app` successfully

braidn commented:
@sherryson how are you installing the gems here? I recommend something like:

`bundle install --path=.bundle`

This will build a clean set of Gems locally for just this project. Then let's see what the bundle exec rake test_app fails with? What is the current failure output?

sherryson commented:
@braidn oooh ok. i was just doing `bundle install`

currently `bundle exec rake test_app` gives me 
```
Generating dummy Rails application...
Setting up dummy database...
bundle exec rake db:migrate VERBOSE=false
rake aborted!
NameError: uninitialized constant Spree::UserRegistrationsController
/Users/sherryson/src/solidus_social/app/controllers/spree/user_registrations_controller_decorator.rb:1:in `<top (required)>'
/Users/sherryson/src/solidus_social/lib/solidus_social/engine.rb:21:in `block in activate'
/Users/sherryson/src/solidus_social/lib/solidus_social/engine.rb:20:in `glob'
/Users/sherryson/src/solidus_social/lib/solidus_social/engine.rb:20:in `activate'
/Users/sherryson/src/solidus_social/spec/dummy/config/environment.rb:5:in `<top (required)>'
/Users/sherryson/.rbenv/versions/2.3.4/bin/bundle:22:in `load'
/Users/sherryson/.rbenv/versions/2.3.4/bin/bundle:22:in `<main>'
Tasks: TOP => db:migrate => environment
(See full trace by running task with --trace)
rake aborted!
Command failed with status (1): [bundle exec rake db:migrate VERBOSE=false...]
/Users/sherryson/src/solidus_social/Rakefile:14:in `block in <top (required)>'
/Users/sherryson/.rbenv/versions/2.3.4/bin/bundle:22:in `load'
/Users/sherryson/.rbenv/versions/2.3.4/bin/bundle:22:in `<main>'
Tasks: TOP => common:test_app
(See full trace by running task with --trace)
```

so i tried creating a db/running migrations separately outside of this rake task but couldn't..

sherryson commented:
got a little help from @braidn (thank you!)

following up on my last comment, we do need Solidus in the Gemfile, since `solidus-core` is only added as a runtime dependency, but instead of using the `master` branch, using the v1.4.2 is probably the right move. 

now the specs are failing (mainly because of older dependencies) and will fix and update here as well

ktheory commented on spree_social.gemspec:
```diff
@@ -21,7 +21,7 @@ Gem::Specification.new do |s|
   s.requirements << 'none'
 
-  s.add_runtime_dependency 'solidus_core', [">= 1.1.0.pre", "<= 1.4.0"]
+  s.add_runtime_dependency 'solidus_core', [">= 1.1.0.pre", "<= 2.0"]
```
```suggestion
  s.add_runtime_dependency 'solidus_core', [">= 1.1.0.pre", "< 2.0"]
```

Did you mean strictly `< 2.0` so we could go to v1.4.2 now, and revisit before 2.0?

If you want to allow v2.0, we should go up to the latest tiny version, `2.0.3`.

braidn replied:
I believe this work is to get this 💎 working with the 1.4 mainline ? Thereby doing incremental bumps to get us there. We also brought the gem version of Solidus in the Gemfile to 1.4.2 so we could be assured our test suite was passing with a 1.4 version of the app.

This causes micro updates which I agree are a little tiring but, due to how large our App is and how different it is from 1.4, being conservative here made sense. Maybe being hyperconservative wasn’t the best move?

ktheory reviewed (commented):
@braidn / @sherryson Is there background on why the constraint on `solids_core` is so pessimistic? Looking through the git history, it's been:

`<= 1.4.0`, `<= 1.3.0.alpha`, and `< 1.2.0`.

Was this out of an abundance of caution to force devs to bump the version before upgrading? Does solidus_social use private APIs we expect to break between point versions?

In short, is there a sufficient and more flexible version constraint than making it a bit wider each time we need to upgrade?

sherryson committed:
use solidus 1.4.2

in order to allow the project to the bundled, we need to use an older version of Solidus, but 2.0 moves to rails 5 so this seems safest for the time being. it also requires deface to be added to the Gemfile.
along with moving to 1.4.2, the gemspec limits solidus to <2.0 (for the same reason as above).
with moving to 1.4.2, some mixins were deprecated and had to be added back in.
we also bump the version of poltergeist use, as 1.5.0 uses a version of phantomJS that is no longer supported

Commit:
Fixes the Spec Suite

The broken specs had two separate causes:

- at some point, the URL helpers for provider-specific authorization were
  changed, either by Devise or by OmniAuth;
- Poltergeist/PhantomJS were misbehaving.

By adjusting the provider-specific login buttons, and switching from PJS to
Selenium Headless Chrome, the specs were fixed.

Some additional specs were broken due to case-sensitivity. These were fixed by
making the expectation case-insensitive regexes instead of case-sensitive
strings.

sherryson commented:
thanks @DanielWright for fixing the specs!

@ktheory @braidn do you think i can merge this into master now?

ktheory commented on spree_social.gemspec:
```diff
@@ -21,7 +21,7 @@ Gem::Specification.new do |s|
   s.requirements << 'none'
 
-  s.add_runtime_dependency 'solidus_core', [">= 1.1.0.pre", "<= 1.4.0"]
+  s.add_runtime_dependency 'solidus_core', [">= 1.1.0.pre", "< 2.0"]
```
Aces ♠️

ktheory reviewed (commented):
LGTM. 👍
